### PR TITLE
Add check to `load_file_to_table_natively` for `storage_integration`

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Currently, we did not check for getting the `storage_integration` param within the `load_file_to_table_natively` method.
Now that we have introduced the `load_options` for snowflake, we can pass `storage_integration` either in `native_support_kwargs` or `load_options`. 

## What is the new behavior?
1. Added condition check for `storage_integration` and added test cases.
2. Added another property `storage_integration` in `SnowflakeLoadOptions` to store `storage_integration`,  since `SnowflakeLoadOptions.copy_options` was intended to be used differently.

## Does this introduce a breaking change?
Nope

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by tatiana — APPROVED** [review]

File: python-sdk/src/astro/databases/snowflake.py [hunk 1]
```diff
@@ -618,9 +618,19 @@ def load_file_to_table_natively(
 
         """
         native_support_kwargs = native_support_kwargs or {}
-        storage_integration = native_support_kwargs.get("storage_integration")
         if not self.load_options:
             self.load_options = SnowflakeLoadOptions()
+
+        storage_integration = native_support_kwargs.get("storage_integration", None)
```
Comment (sunank200) [inline python-sdk/src/astro/databases/snowflake.py:624]: We can create a deprecation warning in another PR.

File: python-sdk/tests/databases/test_snowflake.py [hunk 2]
```diff
@@ -165,3 +165,58 @@ def test_snowflake_load_options_empty():
...
+    with mock.patch("astro.databases.snowflake.SnowflakeDatabase.create_stage"), mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase._copy_into_table_from_stage"
+    ) as _copy_into_table_from_stage, mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase.evaluate_results"
+    ):
+        database.load_file_to_table_natively(
+            source_file=file,
+            target_table=table,
+            native_support_kwargs={"storage_integration": "some_integrations"},
+        )
```
Comment (sunank200) [inline python-sdk/tests/databases/test_snowflake.py:203]: What are we checking as part of this test?

File: python-sdk/tests/databases/test_snowflake.py [hunk 3]
```diff
@@ -165,3 +165,58 @@ def test_snowflake_load_options_empty():
...
+        conn_id=table.conn_id,
+        load_options=SnowflakeLoadOptions(copy_options={"storage_integration": "some_integrations"}),
+    )
+
+    with mock.patch("astro.databases.snowflake.SnowflakeDatabase.create_stage"), mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase._copy_into_table_from_stage"
+    ) as _copy_into_table_from_stage, mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase.evaluate_results"
+    ):
+        database.load_file_to_table_natively(source_file=file, target_table=table)
```
Comment (sunank200) [inline python-sdk/tests/databases/test_snowflake.py:222]: What are we checking as part of this test?

**Review by sunank200 — COMMENTED** [review]
Overall looks good to me but in test we could assert to check the attributes.

File: python-sdk/tests/databases/test_snowflake.py (hunk 2 above)
Comment (utkarsharma2) [reply python-sdk/tests/databases/test_snowflake.py:203]: in the previous test `test_storage_integrations_params` we made sure that if the `storage_integration` is not passed the `load_file_to_table_natively` will raise the ValueError exception. Here we are making sure to pass `storage_integration` via native_support_kwargs don't.

File: python-sdk/tests/databases/test_snowflake.py (hunk 3 above)
Comment (utkarsharma2) [reply python-sdk/tests/databases/test_snowflake.py:222]: in the previous test `test_storage_integrations_params` we made sure that if the `storage_integration` is not passed the `load_file_to_table_natively` will raise the ValueError exception. Here we are making sure to pass `storage_integration` via `SnowflakeLoadOptions` don't.

**Review by sunank200 — APPROVED** [review]

File: python-sdk/tests/databases/test_snowflake.py (hunk 3 above)
Comment (sunank200) [reply python-sdk/tests/databases/test_snowflake.py:222]: Sure

File: python-sdk/tests/databases/test_snowflake.py [hunk 4]
```diff
@@ -165,3 +165,58 @@ def test_snowflake_load_options_empty():
...
+    `native_support_kwargs` or via `SnowflakeLoadOptions` it raises an exception.
+    """
+    table = Table(conn_id="fake-conn")
+    file = File(path="azure://data/homes_main.ndjson")
+    database = SnowflakeDatabase(conn_id=table.conn_id, load_options=SnowflakeLoadOptions())
+    with pytest.raises(DatabaseCustomError) as e:
+        database.load_file_to_table_natively(source_file=file, target_table=table)
+    assert e.match(
+        "Param 'storage_integration' is required. Please pass either in `native_support_kwargs` or"
+        " `SnowflakeLoadOptions.copy_options`"
```
Comment (pankajastro) [inline python-sdk/tests/databases/test_snowflake.py:182]: this should be `SnowflakeLoadOptions.storage_integration` or `SnowflakeLoadOptions.copy_options`?

Comment (utkarsharma2) [reply python-sdk/tests/databases/test_snowflake.py:182]: Yup, will change this. Thanks for pointing out

**Pull request merged**